components/layout_2020/table/layout.rs

```diff
@@ -639,6 +627,18 @@ impl<'a> TableLayout<'a> {
         layout_context: &LayoutContext,
         writing_mode: WritingMode,
     ) -> ContentSizes {
+        // It's not clear whether `inline-size: stretch` allows fixed table mode or not,
+        // we align with Gecko and Blink.
+        // <https://github.com/w3c/csswg-drafts/issues/10937>.
+        self.is_in_fixed_mode = self.table.style.get_table().clone_table_layout() ==
+            TableLayoutMode::Fixed &&
+            !matches!(
+                self.table
+                    .style
+                    .box_size(self.table.style.writing_mode)
+                    .inline,
+                Size::Initial | Size::MaxContent
+            );
```

Comment:
Could this move to the constructor for `TableLayout` as it only seems to rely on `self.table.style`?